python/pyspark/sql/dataframe.py

```diff
@@ -2305,16 +2305,15 @@ def _to_corrected_pandas_type(dt):
     uncorrectly.
     """
     import numpy as np
-    if type(dt) == ByteType:
-        return np.int8
-    elif type(dt) == ShortType:
-        return np.int16
-    elif type(dt) == IntegerType:
-        return np.int32
-    elif type(dt) == FloatType:
-        return np.float32
-    else:
-        return None
+    mappings = {
```

Comment:
We should list up all the data types here. Initially it was in order to correct pandas's inferred type.
Now, in case of empty data, pandas always infers it as `object` and you should rely on this type mapping unlike the intended case before.

See `to_arrow_type` as an example for complete type mapping. You might need to check what Spark -> Python -> pandas type conversion combinations and whitelist it here.